core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/version/VersionCreateRequest.java

```diff
@@ -273,6 +273,33 @@ private Map<ResourceURI, TerminologyResource> fetchResources(ServiceProvider con
 							resourcesToVersion.put(resource.getResourceURI(), resource);
 						}
 					});
+			} else {
+				// otherwise look for derived resources (direct dependencies only)
+				Set<String> derivativeIds = terminologyResource.getDependencies()
+					.stream()
+					.filter(d -> TerminologyResource.DependencyScope.SOURCE_OF.equals(d.getScope()))
+					.map(d -> d.getUri())
+					.filter(uriWithQuery -> !uriWithQuery.hasQueryPart())
+					.map(uriWithQuery -> uriWithQuery.getResourceUri())
+					.filter(uri -> uri.isHead())
+					.map(uri -> uri.getResourceId())
+					.collect(Collectors.toSet());
+				
+				ResourceRequests.prepareSearch()
```

Comment:
The execution of this request can be skipped if there is no `derivativeId` present in the above set.